selftests/unit/test_status_repo.py

```diff
@@ -0,0 +1,67 @@
+from unittest import TestCase
+
+from avocado.core.status import repo, utils
+
+
+class StatusRepo(TestCase):
```

Comment:
Oh, as far as I understand, unit tests should test every single method/function, not just the public API. If a unit test breaks because we changed a private method, that is fine and acceptable in my point of view.

Reply:
It's not often that I add test to private methods, but I agree with @willianrampazzo here.  It's fine, on a refactor, to dish out tests for the dropped internal methods.

Reply:
I understand from TDD that unittests are our first clients. I have some references here too, I can try to collect them. But no worries.

Reply:
I did a quick search on "unit testing private methods," and it seems there is not a consensus about it. There are lots of personal opinions with good arguments on both sides. In this case, either one is fine for me, as it won't hurt having unit tests for private methods.

Reply:
> I did a quick search on "unit testing private methods," and it seems there is not a consensus about it. There are lots of personal opinions with good arguments on both sides. In this case, either one is fine for me, as it won't hurt having unit tests for private methods.

My few cents:

 1. Unit tests classes will be bigger;
  2. We could have the same coverage using only the public ones;
 2. Remove flexibility from us, since that any small change on the internal methods would require changes on the tests;

This "hurts a little", to me. ;)

Reply:
Yep, I got your point and I think it is valid.